eclair-core/src/main/scala/fr/acinq/eclair/Setup.scala

```diff
@@ -122,10 +122,7 @@ class Setup(val datadir: File,
   PortChecker.checkAvailable(serverBindingAddress)
 
   val (bitcoin, bitcoinChainHash) = {
-    val wallet = {
-      val name = config.getString("bitcoind.wallet")
-      if (!name.isBlank) Some(name) else None
-    }
+    val wallet = if (config.hasPath("bitcoind.wallet")) Some(config.getString("bitcoind.wallet")) else None
```

Comment:
How can this be backwards-compatible? It looks like by default we will now end up with `Some("")` where we previously ended up with `None`...

Reply:
By default `config.hasPath("bitcoind.wallet")` will return `false`, so we will end up with `None` like before.

Reply:
I explicitly set `eclair.bitcoind.wallet = ""` in my `eclair.conf` and I suspect other users do as well, so it's safer to keep a check for `isBlank`, isn't it? Otherwise this will break for those users.

Reply:
I was confused about what we're trying to achieve here. Here is a breakdown of what can happen:

- The node operator didn't specify anything in their `eclair.conf`: they were using the default wallet loaded by bitcoin core. After that change, `wallet` will be set to `None` so we'll keep using the default wallet.
- The node operator did specify `wallet = "something"` in their `eclair.conf`: after that change, this will keep using that named wallet.
- The node operator did specify `wallet = ""` in their `eclair.conf`: they were using the default wallet loaded by bitcoin core. After that change, this will now explicitly use a wallet with an empty name.

IIUC the third case is the one that could create issues: if this was actually falling back to a named wallet (because bitcoind had a default loaded named wallet), after that change the user will *not* be using the same wallet anymore? Maybe it's ok, they just need to update their `eclair.conf` to specify the name of the wallet they actually were using (without knowing it)?

Reply:
Oops, you are right. However eclair will fail right at startup with a clear error in that case.

Reply:
That sounds good!